openedx_ledger/admin.py

```diff
@@ -354,3 +388,153 @@ def save_model(self, request, obj, form, change):
                 notes=obj.notes,
                 transaction_of_interest=obj.transaction_of_interest,
             )
+
+
+@admin.register(models.SalesContractReferenceProvider)
+class SalesContractReferenceProviderAdmin(SimpleHistoryAdmin):
+    """
+    Admin configuration for the SalesContractReferenceProvider model.
+    """
+    class Meta:
+        """
+        Metaclass for SalesContractReferenceProviderAdmin.
+        """
+
+        model = models.SalesContractReferenceProvider
+        fields = '__all__'
+
+    search_fields = ('name', 'slug',)
+    list_display = ('name', 'slug',)
+
+
+class DepositAdminCreateForm(forms.ModelForm):
+    """
+    Form that allows users to enter deposit quantities in dollars
+    instead of cents.
+    """
+    class Meta:
+        model = models.Deposit
+        fields = [
+            'ledger',
+            'quantity_usd_input',
+            'sales_contract_reference_id',
+            'sales_contract_reference_provider',
+        ]
+
+    quantity_usd_input = forms.FloatField(
+        required=True,
+        help_text=(
+            "Amount of deposit in US Dollars. Remember to increase the related policy spend limits to take advantage "
+            "of this new added value."
+        ),
+    )
+
+
+class DepositAdminChangeForm(forms.ModelForm):
+    """
+    Form for reading and changing only the allowed fields of an existing Deposit record.
+    """
+    class Meta:
+        model = models.Deposit
+        fields = [
+            'ledger',
+            'transaction',
+            'desired_deposit_quantity',
+            'sales_contract_reference_id',
+            'sales_contract_reference_provider',
+        ]
+
+
+@admin.register(models.Deposit)
+class DepositAdmin(SimpleHistoryAdmin):
+    """
+    Admin configuration for the Deposit model.
+    """
+    form = DepositAdminCreateForm
+
+    _readonly_fields = [
+        'get_quantity_usd',
+        'uuid',
+        'transaction',
+        'desired_deposit_quantity',
+        'created',
+        'modified',
+    ]
+
+    list_display = (
+        'uuid',
+        'get_ledger_uuid',
+        'get_quantity_usd',
+        'sales_contract_reference_id',
+        'created',
+        'modified',
+    )
+
+    # There is no low-cardinality field to filter on for Deposits.
+    # list_filter = ()
```

Comment:
[nit] Should this still be here?

Reply:
yea, it's just to explain why the list_filter is empty.